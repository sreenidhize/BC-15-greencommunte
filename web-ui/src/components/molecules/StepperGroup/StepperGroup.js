import { makeStyles } from "@material-ui/core";
import React from "react";
import Stepper from "../Stepper/Stepper";

const useStyles = makeStyles({
  root: {
    display: "flex",
    justifyContent: "space-evenly",
    // width: "700px",
  },
  items: {
    display: "flex",
    alignItems: "center",
  },
});

const StepperGroup = (props) => {
  const classes = useStyles();
  return (
    <div className={classes.root} data-testid="stepper-group">
      <Stepper  number="1"  label="Your Location" isVisited={props.item1}/>
      <Stepper number="2"  label="Job Location" isVisited={props.item2} />
      <Stepper number="3" label="Your skills" isVisited={props.item3} />
    </div>
  );
};

export default StepperGroup;
