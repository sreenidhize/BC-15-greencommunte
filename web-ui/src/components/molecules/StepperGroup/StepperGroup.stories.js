import React from 'react';
import StepperLabel from './StepperGroup';

export default {
    title: 'molecules/StepperGroup',
    component: StepperLabel,
};

const Template = (args) => <StepperLabel {...args} />;
 
export const Primary = Template.bind({});

Primary.args = {
    item1: true,
    item2: true,
    item3: true,
};
