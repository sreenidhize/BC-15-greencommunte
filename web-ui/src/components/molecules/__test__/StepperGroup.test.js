import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import StepperGroup from "../StepperGroup/StepperGroup";


describe("StepperGroup test", () => {
    it("Matches the snapshot", () => {
      const { container } = render(
        <StepperGroup item1={false} item2={false} item3={false} />
      );
      expect(container).toMatchSnapshot();
    });
  
    test("Test inactive items in Stepper Group  render", () => {
      const { getByTestId } = render(<StepperGroup item1={false} item2={false} item3={false} />);
      expect(getByTestId("stepper-group")).toBeInTheDocument();
    });
  
    test("Test active items in Stepper Group render", () => {
      const { getByTestId } = render(<StepperGroup item1={true} item2={true} item3={true} /> );
      expect(getByTestId("stepper-group")).toBeInTheDocument();
    });
  });