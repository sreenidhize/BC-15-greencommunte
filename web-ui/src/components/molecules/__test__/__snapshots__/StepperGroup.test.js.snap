// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepperGroup test Matches the snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-1"
    data-testid="stepper-group"
  >
    <div
      data-testid="stepper"
      style="display: flex; align-items: center;"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-unvisited-3 MuiAvatar-colorDefault"
        data-testid="label-number"
      >
        1
      </div>
      <p
        class="MuiTypography-root makeStyles-root-5 MuiTypography-body1"
        data-testid="stepper-label"
        style="color: rgb(95, 115, 129); font-size: 18px; margin-right: 15px;"
      >
        Your Location
      </p>
    </div>
    <div
      data-testid="stepper"
      style="display: flex; align-items: center;"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-unvisited-3 MuiAvatar-colorDefault"
        data-testid="label-number"
      >
        2
      </div>
      <p
        class="MuiTypography-root makeStyles-root-5 MuiTypography-body1"
        data-testid="stepper-label"
        style="color: rgb(95, 115, 129); font-size: 18px; margin-right: 15px;"
      >
        Job Location
      </p>
    </div>
    <div
      data-testid="stepper"
      style="display: flex; align-items: center;"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-unvisited-3 MuiAvatar-colorDefault"
        data-testid="label-number"
      >
        3
      </div>
      <p
        class="MuiTypography-root makeStyles-root-5 MuiTypography-body1"
        data-testid="stepper-label"
        style="color: rgb(95, 115, 129); font-size: 18px; margin-right: 15px;"
      >
        Your skills
      </p>
    </div>
  </div>
</div>
`;
